vllm/entrypoints/llm.py

```diff
@@ -218,3 +218,9 @@ def _run_engine(self, use_tqdm: bool) -> List[RequestOutput]:
         # its previous requests.
         outputs = sorted(outputs, key=lambda x: int(x.request_id))
         return outputs
+
+    def __del__(self):
+        del self.llm_engine.driver_worker.model_runner
+        del self.llm_engine.driver_worker
```

Comment:
We will also have to handle the case where there are multiple gpus.

In this case `llm_engine` will also have a `workers` list